From 45aef8f3d4a16b0273648de81763e8fdab118138 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Wed, 16 Feb 2011 21:13:28 +0200
Subject: [PATCH 0303/1092] cbus: retu: wdt: save dev in retu_wdt_dev

commit 6b8074b00d90b191227dc875b90b272c51f7d6eb
(cbus: Make retu watchdog behave like a standard
Linux watchdog) introduced struct retu_wdt_dev
with a field to save struct device *dev, but never
got the point of saving the pointer into that
structure.

Fix this very old bug so we can actually start
passing child device pointers to retu read/write
operations.

Signed-off-by: Felipe Balbi <balbi@ti.com>
---
 drivers/cbus/retu-wdt.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/cbus/retu-wdt.c b/drivers/cbus/retu-wdt.c
index 423216c..f6ff9ad 100644
--- a/drivers/cbus/retu-wdt.c
+++ b/drivers/cbus/retu-wdt.c
@@ -258,6 +258,7 @@ static int __init retu_wdt_probe(struct platform_device *pdev)
 	if (!wdev)
 		return -ENOMEM;
 
+	wdev->dev = &pdev->dev;
 	wdev->users = 0;
 
 	ret = device_create_file(&pdev->dev, &dev_attr_period);
-- 
1.7.11.2

